prowler/lib/check/models.py

```diff
@@ -405,16 +405,20 @@ class Check_Report:
     status: str
     status_extended: str
     check_metadata: CheckMetadata
+    resource_metadata: dict
     resource_details: str
     resource_tags: list
     muted: bool
 
-    def __init__(self, metadata):
+    def __init__(self, metadata, resource_metadata=None):
```

Comment:
```suggestion
    def __init__(self, metadata, resource=None):
```

I like more to call it `resource` since we are passing the object itself.